client/src/main/scala/org/apache/celeborn/client/ChangePartitionManager.scala

```diff
@@ -47,8 +47,14 @@ class ChangePartitionManager(
   // shuffleId -> (partitionId -> set of ChangePartition)
   private val changePartitionRequests =
     JavaUtils.newConcurrentHashMap[Int, ConcurrentHashMap[Integer, JSet[ChangePartitionRequest]]]()
```

Comment:
Hi, I tested this patch internally, but still saw 1 shuffle read < shuffle write with small margin. And we are still looking into it.
```
stage 0 ----Shuffle Write Size / Records: 20.8 TiB / 1965259729
stage 1 ---- Shuffle Read Size / Records: 20.8 TiB / 1965259725
```
Do we need to use concurrent hash set for the ChangePartitionRequest as well? or any other places that could have caused the small shuffle loss? thanks